v3/src/data-interactive/resource-parser.ts

```diff
@@ -98,7 +101,13 @@ export function resolveResources(
 
   if (resourceSelector.component) {
     // TODO Get tile by name?
-    result.component = document.content?.getTile(resourceSelector.component)
+    const { component } = resourceSelector
+    // We look for every possible v3 id the component might have (because each tile type has a different prefix).
+    // Is there a better way to do this?
+    const possibleIds =
+      [component, toV3TileId(component), ...getTilePrefixes().map(prefix => toV3Id(prefix, component))]
+    const componentId = possibleIds.find(id => document.content?.getTile(id))
+    if (componentId) result.component = document.content?.getTile(componentId)
```

Comment:
I'm inclined to say that this is fine for now. If we wanted to do better we could put the responsibility on the `DocumentContent`, which could maintain a mapping from prefix-less id to tile, for instance, but that can be left as a potential optimization for another day.